### PR TITLE
xml2js: Allow parsing a NodeJS buffer

### PR DESCRIPTION
**Pull request by DavidM77**
Please fill in this template.

- [X] Prefer to make your PR against the `types-2.0` branch.
- [X] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

If changing an existing definition:
- [X] Provide a URL to  documentation or source code which provides context for the suggested changes: <https://www.npmjs.com/package/xml2js#simple-as-pie-usage>
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

File: xml2js/index.d.ts [hunk 1]
```diff
@@ -70,5 +70,9 @@ declare namespace xml2js {
         chunkSize?: number;
         cdata?: boolean;
     }
+
+    interface convertableToString {
```
Comment (sandersn) [inline xml2js/index.d.ts]: types should start with an uppercase letter

**Comment by sandersn** [comment]
Supporting anything with `toString` is *almost* everything in Javascript. Can the authors say whether this is a good idea? @michelsalib @jasonrm @ccurrens

**Comment by michelsalib** [comment]
Hmmm what you want is to support parsing of buffer... so why not doing:
`string|Buffer`

**Comment by michelsalib** [comment]
Actually, xml2js is going to call the toString function... so this fix is not that a bad idea: https://github.com/Leonidas-from-XIV/node-xml2js/blob/master/src/parser.coffee#L235

**Pull request merged**